### PR TITLE
Documentation for preemption

### PR DESCRIPTION
**Pull request by preetapan**
Reorganized the internals docs to make scheduling a new section and added preemption in there. 

Also added docs for scheduler config API that I missed adding along with the code earlier.

### CODE REVIEW

File: website/source/api/operator.html.md [hunk 1]
```diff
@@ -328,3 +328,97 @@ $ curl \
...
+```json
+{
+  "PreemptionConfig": {
+    "EnablePreemption": false
+  }
+}
+```
+
+- `PreemptionConfig` `(PreemptionConfig)` - Options to enable preemption for various schedulers.
+ - `SystemSchedulerEnabled`(bool:true) - Specifies whether preemption for system jobs is enabled. Note that
```
Comment (dadgar) [inline website/source/api/operator.html.md]:
```suggestion
 - `SystemSchedulerEnabled` (`bool`: `true`) - Specifies whether preemption for system jobs is enabled. Note that
```

Comment (dadgar) [reply website/source/api/operator.html.md]: This page has pretty inconsistent formatting on fields/types/defaults

File: website/source/docs/internals/scheduling/scheduling.html.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: "docs"
 page_title: "Scheduling"
-sidebar_current: "docs-internals-scheduling"
+sidebar_current: "docs-internals-scheduling-internals"
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/scheduling.html.md]: Delete the copy that is now in the index.html?

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+…
+
+Scheduler dry-run:
+- All tasks successfully allocated.
+
+Preemptions:
+
+Alloc ID                              Job ID    Task Group
+ddef9521                              my-batch   analytics
+
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: delete blank

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# Integration with Nomad plan
+
+`nomad plan` allows operators to dry run the scheduler. If the scheduler determines that
+preemption is necessary to place the job, it shows additional information in the CLI output for
+`nomad plan` as seen below.
+
+```sh
+$ nomad plan example.nomad
+…
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: I would show the full plan. It isn't that long and as it is now, it is somewhat confusing

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+to how closely they fit the job's required capacity. For example, if the `75` priority job needs 1GB disk and 2GB memory, Nomad will preempt
+allocations `a1`, `a2` and `a4` to satisfy those requirements.
+
+# Preemption Visibility
+
+Operators can use the [allocation API](/api/allocations.html#read-allocation) to get visibility into whether an allocation has been preempted.
+Preempted allocations will have their DesiredStatus set to “evict”. The `Allocation` object in the API also has two additional fields related to
+preemption.
+
+- PreemptedAllocs - This field is set on an allocation that caused preemption. It contains the allocation ids of allocations
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: Use code ticks for the field names

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+job remain in the pending state until sufficient resources become available - regardless of the defined priority.
+This leads to priority inversion, where a low priority task can prevent high priority tasks from completing.
+
+Nomad 0.9 brings preemption capabilities to system jobs. The Nomad scheduler will evict lower priority running allocations
+to free up capacity for new allocations resulting from relatively higher priority jobs, sending evicted allocations back
+into the plan queue.
+
+# Details
+
+Preemption is enabled by default in Nomad 0.9. Operators can use the [scheduler config][/api/operator.html#update-scheduler-configuration] API endpoint to disable preemption.
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: The link doesn't render. The URI should be in () not []

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Nomad uses the [job priority](/docs/job-specification/job.html#priority) field to determine what running allocations can be preempted.
+In order to prevent a cascade of preemptions due to jobs close in priority being preempted, only allocations from jobs with a priority
+delta of more than 10 from the job needing placement are chosen.
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]:
```suggestion
delta of more than 10 from the job needing placement are eligible for preemption.
```

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+eligible to be preempted because they are of a lower priority. Allocations from the `cache` job will never be preempted because its priority value `70`
+is lesser than the required delta of `10`.
+
+Allocations are selected starting from the lowest priority, and scored according
+to how closely they fit the job's required capacity. For example, if the `75` priority job needs 1GB disk and 2GB memory, Nomad will preempt
+allocations `a1`, `a2` and `a4` to satisfy those requirements.
+
+# Preemption Visibility
+
+Operators can use the [allocation API](/api/allocations.html#read-allocation) to get visibility into whether an allocation has been preempted.
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: What about the CLI?

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+Scheduler dry-run:
+- All tasks successfully allocated.
+
+Preemptions:
+
+Alloc ID                              Job ID    Task Group
+ddef9521                              my-batch   analytics
+
+```
+
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: Add more to here explaining that there is no guarantee that the allocation shown will be the allocation preempted at run time but will give the operator a sample of the type of allocations that will be prompted

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+$ nomad plan example.nomad
+…
+
+Scheduler dry-run:
+- All tasks successfully allocated.
+
+Preemptions:
+
+Alloc ID                              Job ID    Task Group
+ddef9521                              my-batch   analytics
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: May be worth using an example with more than one alloc

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+```sh
+$ nomad plan example.nomad
+…
+
+Scheduler dry-run:
+- All tasks successfully allocated.
+
+Preemptions:
+
+Alloc ID                              Job ID    Task Group
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: We should probably have a namespace column as well?

File: website/source/docs/internals/scheduling/preemption.html.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+layout: "docs"
+page_title: "Preemption"
+sidebar_current: "docs-internals-scheduling-preemption"
+description: |-
+  Learn about how preemption works in Nomad.
+---
+
+# Preemption
+
+Preemption refers to the temporary interruption of a computing task, “without requiring its cooperation,
```
Comment (dadgar) [inline website/source/docs/internals/scheduling/preemption.html.md]: Little textbooky:

```
Preemption allows Nomad to kill existing allocations in order to place allocations for a higher priority job. The evicted allocation is temporary displaced until the cluster has capacity to run it. This capability allows operators to run high priority jobs even under resource contention across the cluster.
```

File: website/source/api/operator.html.md [hunk 13]
```diff
@@ -328,3 +328,97 @@ $ curl \
...
+| `GET`  | `/operator/scheduler/configuration` | `application/json` |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes), and
+[required ACLs](/api/index.html#acls).
+
+| Blocking Queries | Consistency Modes | ACL Required    |
+| ---------------- | ----------------- | --------------- |
+| `NO`             | `none`            | `operator:read` |
```
[the remark this answers is not in the document]
Comment (preetapan) [reply website/source/api/operator.html.md]: I copied this from autopilot docs, so looks like those are wrong as well!

File: website/source/docs/internals/scheduling/preemption.html.md (hunk 11 above)
Comment (preetapan) [reply website/source/docs/internals/scheduling/preemption.html.md]: will address this in a followup ticket, we currently don't show namespace in the CLI

File: website/source/docs/internals/scheduling/scheduling.html.md (hunk 2 above)
Comment (preetapan) [reply website/source/docs/internals/scheduling/scheduling.html.md]: index.html is only a small snippet of content that links to the scheduling and preemption pages.

File: website/source/api/operator.html.md [hunk 14]
```diff
@@ -420,5 +420,5 @@ The table below shows this endpoint's support for
 ```
 
 - `PreemptionConfig` `(PreemptionConfig)` - Options to enable preemption for various schedulers.
- - `SystemSchedulerEnabled`(bool:true) - Specifies whether preemption for system jobs is enabled. Note that
-         if this is set to true, then system jobs can preempt any other jobs.
\ No newline at end of file
+ - `SystemSchedulerEnabled` (`bool`: `true`) - Specifies whether preemption for system jobs is enabled. Note that
```
Comment (dadgar) [inline website/source/api/operator.html.md]: I guess everything else is just doing ticks around the entire parenthesis. Was thinking about how the job spec does it.

**Pull request merged**